tests/cmdline/commands/test_group.py

```diff
@@ -134,24 +134,146 @@ def test_delete(self, run_cli_command):
         with pytest.raises(exceptions.NotExistent):
             orm.load_group(label='group_test_delete_02')
 
-        # check nodes still exist
         for pk in node_pks:
             orm.load_node(pk)
 
-        # delete the group and the nodes it contains
+        # 4) Delete the group and the nodes it contains
         group = orm.load_group(label='group_test_delete_03')
         group.add_nodes([node_01, node_02])
         result = run_cli_command(
             cmd_group.group_delete, ['--force', '--delete-nodes', 'group_test_delete_03'], use_subprocess=True
         )
 
-        # check group and nodes no longer exist
         with pytest.raises(exceptions.NotExistent):
             orm.load_group(label='group_test_delete_03')
         for pk in node_pks:
             with pytest.raises(exceptions.NotExistent):
                 orm.load_node(pk)
 
+        # 5) Should delete an empty group even if --delete-nodes option is passed
+        group = orm.Group(label='group_test_delete_04').store()
+        result = run_cli_command(
+            cmd_group.group_delete, ['--force', '--delete-nodes', 'group_test_delete_04'], use_subprocess=True
+        )
+        with pytest.raises(exceptions.NotExistent):
+            orm.load_group(label='group_test_delete_04')
+
+        # 6) Should raise if a group does not exist
+        result = run_cli_command(
+            cmd_group.group_delete, ['--force', 'non_existent_group'], raises=True, use_subprocess=True
+        )
+
+        # 7) Should delete multiple groups
+        orm.Group(label='group_test_delete_05').store()
+        orm.Group(label='group_test_delete_06').store()
+        result = run_cli_command(
+            cmd_group.group_delete, ['--force', 'group_test_delete_05', 'group_test_delete_06'], use_subprocess=True
+        )
+        with pytest.raises(exceptions.NotExistent):
+            orm.load_group(label='group_test_delete_05')
+        with pytest.raises(exceptions.NotExistent):
+            orm.load_group(label='group_test_delete_06')
+
+        # 8) Should raise if both groups and query options are passed
+        orm.Group(label='group_test_delete_07').store()
+        result = run_cli_command(
+            cmd_group.group_delete,
+            ['--force', 'group_test_delete_01', '--all-users', 'label==group_test_delete_07'],
+            raises=True,
+            use_subprocess=True,
+        )
+        result = run_cli_command(
+            cmd_group.group_delete,
+            ['--force', 'group_test_delete_01', '--user', 'non_existent', 'label==group_test_delete_07'],
```

Comment:
Note that `group_test_delete_01` does not exist I think, so it may just be failing because that group is not found. Best change it for a group that actually exists _and_ add an explicit check for the error message in all of these.

Reply:
No wonder they all passed at the first run, I was happy to see a miracle . 
Now I see they were all wrong, hahaha. 
fixed now.